common/src/dynamic_vault/pda.rs

```diff
@@ -0,0 +1,32 @@
+use solana_sdk::pubkey::Pubkey;
+
+pub fn derive_vault_key(mint: Pubkey) -> Pubkey {
+    Pubkey::find_program_address(
+        &[
+            dynamic_vault::seed::VAULT_PREFIX.as_ref(),
+            mint.as_ref(),
+            dynamic_vault::get_base_address().as_ref(),
+        ],
+        &dynamic_vault::ID,
+    )
+    .0
+}
+
+pub fn derive_token_vault_key(vault: Pubkey) -> Pubkey {
+    Pubkey::find_program_address(
+        &[
+            dynamic_vault::seed::TOKEN_VAULT_PREFIX.as_ref(),
+            vault.as_ref(),
+        ],
+        &dynamic_vault::ID,
+    )
+    .0
+}
+
+pub fn derive_lp_mint_key(vault: Pubkey) -> Pubkey {
```

Comment:
I think there are some token mint that has LP is not PDA 
- SOL
- USDC
- USDT
Could you add some special cases in here? can use with feature like mainnet, devnet

Reply:
Fixed